tracer/test/Datadog.Trace.TestHelpers.AutoInstrumentation/AspNetCoreTestFixture.cs

```diff
@@ -134,7 +140,8 @@ public async Task TryStartApp(TestHelper helper, bool? enableSecurity = null, st
                 }
             }
 
-            await EnsureServerStarted(sendHealthCheck);
+            await EnsureServerStarted(sendHealthCheck, waitForHealthcheckSpan);
+            Agent.SpanFilters.Add(IsNotServerLifeCheck);
```

Comment:
Why did you move this? 🤔  I would expect that we need to add the filter _before_ waiting for the health check span no, otherwise won't that break things?

Reply:
So refering to the [previous comment](https://github.com/DataDog/dd-trace-dotnet/pull/4654/files#r1376415688) , it's because here we want to be able to wait for the alive-check span to make sure it's behind us and no concurrent request will interfere in the OverheadController and then we add the filter..

before this mechanism, there was a lot of flake, the test would sometimes show spans without api security because api security info had been added the alive check span... 
so now we want to :
- ensure server started and that the alive check span has been processed
- to wait for alive check span we need to  stop filtering it out in the function `WaitForSpans`
- and then add the filter back so that it works normally..